content/documents/dev/building_haiku_ubuntu_linux_step_step.html

```diff
@@ -1,92 +1,82 @@
 +++
 type = "article"
 title = "Building Haiku on Ubuntu Linux, Step by Step"
-date = "2007-07-21T04:11:17.000Z"
-tags = ["obsolete"]
+date = "2019-03-05T11:19:23.0530Z"
 +++
+# Architecture
 
+To check whether your target architecture is support visit [Port Status](https://www.haiku-os.org/guides/building/port_status).
 
-    <div class="field field-type-text field-field-body">
-    <div class="field-items">
-            <div class="field-item odd">
-                    <div class="box-info">
-<strong>2010-January: This document is now obsolete. We are in the process of consolidating and re-organizing the website documentation. For now please refer to the <a href="/guides">in-progress website documentation</a>.</strong>
-</div>
+# Pre-requisite Software
 
-<p><a href="http://haiku-os.org/documents/dev/getting_linux_developer_tools">Another article on this site</a> already describes the basics of building Haiku on Linux. Since my distribution of choice is Ubuntu, I decided to get Haiku building on it and then provide a detailed step-by-step guide for others to follow.
-</p>
+You need some tools in order to build Haiku. See [Pre-requisite Software](https://www.haiku-os.org/guides/building/pre-reqs) and install the tools mentioned there.
 
-<p>
-	I performed these steps on a fresh Ubuntu 7.04 (Feisty Fawn) install, but they should be similar for other versions and probably the same for most Debian-based distributions.
-</p>
+# Getting the source code
 
+To download Haiku's source anonymously, enter
 
-<!--more-->
+```sh
+git clone https://review.haiku-os.org/buildtools #needed for building
+git clone https://review.haiku-os.org/haiku #actual source
+```
 
+If you'd like to learn more about downloading through git and git workflow, visit [Getting the source code](https://www.haiku-os.org/guides/building/get-source-git).
 
-<ol>
-<li>Install the needed packages for building:
+# Compiling for x86_64
+x86_64 Compiler Toolset
+=======================
 
-<pre class="terminal">sudo apt-get install subversion yasm autoconf automake texinfo flex bison gawk build-essential</pre></li>
+Building the x86_64 compiler toolset is quite easy and involves generating GCC
+binaries for your platform. For a complete list of flags for the configure
+script, see Haiku's [Configure Options]().
```

Comment:
The missing URL should be https://www.haiku-os.org/guides/building/configure , I suppose.

Reply:
@humdingerb I think I've added the missing URL. Please tell me if I should change anything else.

Thank you

Reply:
Thanks, looks good. Let's merge!